### DISCUSSION TITLE
how to inference based multisports pretrained model ??

### DISCUSSION THREAD

**Discussion by Eurususu** [open-mmlab/mmaction2#2596] [Q&A]
I used given link [https://download.openmmlab.com/mmaction/v1.0/detection/slowfast/slowfast_kinetics400-pretrained-r50_8xb16-4x16x1-8e_multisports-rgb/slowfast_kinetics400-pretrained-r50_8xb16-4x16x1-8e_multisports-rgb_20230320-af666368.pth](url) to download Multisports model .

Then I wannna inference with the model, command like:

> python3 demo/demo_spatiotemporal_det.py --config configs/detection/slowfast/slowfast_kinetics400-pretrained-r50_8xb16-4x16x1-8e_multisports-rgb.py --checkpoint weights/slowfast_kinetics400-pretrained-r50_8xb16-4x16x1-8e_multisports-rgb_20230320-af666368.pth --det-config demo/demo_configs/faster-rcnn_r50_fpn_2x_coco_infer.py --det-checkpoint weights/faster_rcnn_r50_fpn_2x_coco_bbox_mAP-0.384_20200504_210434-a5d8aa15.pth --det-score-thr 0.45 --action-score-thr 0.5 --label-map tools/data/multisports/label_map.txt --predict-stepsize 8 --output-stepsize 4 --output-fps 8 data/multisports/test/basketball/v_4Aoy9-NoY2I_c600.mp4 work_dirs/slowfast_kinetics400-pretrained-r50_8xb16-4x16x1-8e_multisports-rgb/test.mp4

The result I get is a mess!
![截图 2023-07-17 15-08-21](https://github.com/open-mmlab/mmaction2/assets/62414599/529416f3-5286-44c9-bc70-01bfa75b21e1)

But I have seen in multisports official webside, the result in video is
![截图 2023-07-17 15-12-16](https://github.com/open-mmlab/mmaction2/assets/62414599/2bcce050-1e14-4f6b-becf-65408a8e3d81)

**Comment by cir7** [comment]
Hi, @Eurususu 
The SlowFast-Det model takes a two-stage action detection, which would recognize the action of each person in the video. That's why your inference result has so many detected actions, using a higher threshold for classification would make it simpler.
The below image is from the official website and looks like a visualization of ground truth.  
Basketball-related actions are pretty hard in the dataset, you can refer to the table in the table from paper:
![image](https://github.com/open-mmlab/mmaction2/assets/33249023/5454bd8a-2ba2-41f0-9f0a-53346b450fa4)
aerobic videos have fewer people, related actions are easier, and have a better performance, you can try on it.

**Comment by Eurususu** [comment]
Thanks for answering!!

I have tried higher threshold, the boxes is less, but I still can't get the result as showed on multisports official. 

This is threshold 0.9 for action detection.
![截图 2023-07-17 16-18-49](https://github.com/open-mmlab/mmaction2/assets/62414599/ccd04091-f44a-4ce5-8f8e-dfce3fccb342)

what can I do to get the good result?

**Reply by Eurususu** [reply]
@cir7

**Reply by cir7** [reply]
The video from the official website looks like a visualization of annotation, the model can't achieve the performance.
